.github/workflows/ci.yaml

```diff
@@ -0,0 +1,47 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+  workflow_dispatch: {}
+
+env:
+  DOCKER_USR: ${{ secrets.UPBOUND_ROBOT_USR }}
+
+jobs:
+  configuration:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0
+      # The tagger step uses the same logic in the build submodule to generate package tag
+      # https://github.com/upbound/build/blob/4f64913157a952dbe77cd9e05457d9abe695a1d4/makelib/common.mk#L193
+      - name: Set tag
+        run: echo "::set-output name=VERSION_TAG::$(git describe --dirty --always --tags | sed 's/-/./2' | sed 's/-/./2' )"
+        id: tagger
+      - name: Login to Docker
+        uses: docker/login-action@v1
+        if: env.DOCKER_USR != ''
+        with:
+          registry: registry.upbound.io
+          username: ${{ secrets.UPBOUND_ROBOT_USR }}
+          password: ${{ secrets.UPBOUND_ROBOT_PSW }}
+      - name: Build
+        uses: crossplane-contrib/xpkg-action@v0.2.0
+        with:
+          channel: stable
+          version: current
+          command: build configuration -f ${{ github.workspace }} --name package.xpkg --ignore '.github/*,.github/*/*,examples/*,hack/*'
+      - name: Push
+        uses: crossplane-contrib/xpkg-action@v0.2.0
+        with:
+          command: push configuration -f package.xpkg registry.upbound.io/upbound/platform-ref-azure:${{ steps.tagger.outputs.VERSION_TAG }}
```

Comment:
@natereid72 was wondering how the right version is used to build/push the package.  Note on this line, where we are pushing the configuration to the upbound registry, it's using a version of `steps.tagger.outputs.VERSION_TAG`.

Going up to line 24, the tagger steps computes the version and stores it into a variable called `VERSION_TAG`.  The way it computes the version is by using `git describe`, just like our build submodule does in https://github.com/upbound/build/blob/master/makelib/common.mk#L209-L219, which relies on tags in the repository to generate a version based off that tag.  So once we merge this, we should tag the repo (using the new tag action) as `v0.1.0`, then the CI will pick that tag up in its version calculation and voila!